security/integration/main.go

```diff
@@ -143,77 +150,138 @@ func initializeIntegrationTest(cmd *cobra.Command, args []string) error {
-	if s, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
-		secretWaitTime); err != nil {
+	s, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
+		secretWaitTime)
+	if err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" is correctly created`)
+	}
 
-		expectedID := fmt.Sprintf("spiffe://cluster.local/ns/%s/sa/default", s.GetNamespace())
-		examineSecret(s, expectedID)
+	glog.Info(`Secret "istio.default" is correctly created`)
+	_, _, _, err = examineSecret(s)
+	if err != nil {
+		return err
 	}
 
 	// Delete the secret.
 	do := &metav1.DeleteOptions{}
 	if err := opts.clientset.CoreV1().Secrets(opts.namespace).Delete("istio.default", do); err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" has been deleted`)
 	}
+	glog.Info(`Secret "istio.default" has been deleted`)
 
 	// Test that the deleted secret is re-created properly.
 	if _, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
 		secretWaitTime); err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" is correctly re-created`)
 	}
+	glog.Info(`Secret "istio.default" is correctly re-created`)
+
+	return nil
+}
 
-	// Delete pods
-	err = utils.DeletePod(opts.clientset, opts.namespace, self_signed_ca_pod.GetName())
+func cleanUpSelfSignedCATests() {
+	glog.Info("Removing pods: istio-ca-self")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "istio-ca-self")
+	glog.Info("Removing secrets: istio.default")
+	_ = utils.DeleteSecrets(opts.clientset, opts.namespace, "istio.default")
+}
+
+func runCertificatesRotationTests() error {
+	// Create Istio CA with short lifetime certificates
+	_, err := utils.CreatePod(opts.clientset, opts.namespace,
+		fmt.Sprintf("%v/istio-ca-short:%v", opts.containerHub, opts.containerTag), "istio-ca-short")
 	if err != nil {
-		return fmt.Errorf("failed to delete Istio CA (error: %v)", err)
+		return fmt.Errorf("failed to deploy Istio CA istio-ca-short")
 	}
+	glog.Infof("Successfully created Istio CA pod(istio-ca-short) with the short lifetime certificates")
 
-	return nil
+	initialSecret, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
+		secretWaitTime)
+	if err != nil {
+		return err
+	}
+
+	initialKey, initialCert, initialRoot, err := examineSecret(initialSecret)
+	if err != nil {
+		return err
+	}
+
+	maxRetry := certValidateRetry
+	term := certValidationInterval
+
+	for i := 0; i < maxRetry; i++ {
+		if i > 0 {
+			glog.Infof("Checking certificate rotation in %v seconds", term)
+			time.Sleep(time.Duration(term) * time.Second)
+			term = term * 2
+		}
+
+		secret, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
+			secretWaitTime)
+		if err != nil {
+			continue
+		}
+
+		key, cert, root, err := examineSecret(secret)
+		if err != nil {
+			return err
+		}
+
+		if string(initialRoot) != string(root) {
```

Comment:
why not `bytes.Equal(initialRoot, root)`?

Reply:
Replaced